pkg/core/xds/merge.go

```diff
@@ -0,0 +1,112 @@
+package xds
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch/v5"
+)
+
+// indicates that all slices that start with this prefix will be appended, not replaced
+const appendSlicesPrefix = "Append"
+
+func MergeConfs(confs []interface{}) (interface{}, error) {
+	if len(confs) == 0 {
+		return nil, nil
+	}
+
+	resultBytes := []byte{}
+	for _, conf := range confs {
+		confBytes, err := json.Marshal(conf)
+		if err != nil {
+			return nil, err
+		}
+		if len(resultBytes) == 0 {
+			resultBytes = confBytes
+			continue
+		}
+		resultBytes, err = jsonpatch.MergePatch(resultBytes, confBytes)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	result, err := newConf(reflect.TypeOf(confs[0]))
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(resultBytes, result); err != nil {
+		return nil, err
+	}
+
+	clearAppendSlices(reflect.ValueOf(result)) // clear appendable slices, so we won't duplicate values of the last conf
+	for _, conf := range confs {
+		appendSlices(reflect.ValueOf(result), reflect.ValueOf(&conf).Elem()) // call .Elem() to unwrap interface{}
+	}
+
+	v := reflect.ValueOf(result).Elem().Interface()
+
+	return v, nil
+}
+
+func clearAppendSlices(val reflect.Value) {
```

Comment:
if you create a method like:
```go
func mustStructOrPtrToStruct(val reflect.Value) reflect.Value {
	switch val.Kind() {
	case reflect.Struct:
		return val
	case reflect.Pointer:
		if ptrVal := val.Elem(); ptrVal.Kind() == reflect.Struct {
			return ptrVal
		}
	}
	panic("expected struct or pointer to a struct")
}
```
then you can use it at the beginning of `clearAppendSlices` and `appendSlices`

Reply:
Good idea! Although I changed it a bit and also I had to handle interface.